contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -1939,6 +1959,35 @@ check_nullable_identity_constraint(RangeVar *relation, ColumnDef *column)
 						relation->relname)));
 }
 
+/*
+ * Returns true iff the function is a system function. For Babelfish, this means searching in the "sys"
+ * schema, but the "sys" schema is meant to be opaque to customers, so to match SQL Server behavior
+ * we assume that it will never be schema-qualified in order to be true.
+ */
+static bool
+isTsqlSystemFunc(FuncCall *fc)
+{
+	List *name_to_search;
+	ObjectWithArgs *owa = makeNode(ObjectWithArgs);
+	if (list_length(fc->funcname) == 1)
+	{
+		/* explicitly search in the "sys" schema */
+		name_to_search = list_make2(makeString("sys"), linitial(fc->funcname));
```

Comment:
What if user created a function with the same name as a system function in dbo schema ? Do we want to allow them in defaults ?

Reply:
Can we add tests for it as well.

Reply:
In SQL Server, functions with the same name as system functions cannot be used as column defaults in temp tables:
```
1> create table #t1 (a int default dbo.isjson('hi'))
2> go
Msg 342, Level 16, State 1, Server sql1, Line 1
Column "dbo" is not allowed in this context, and the user-defined function or aggregate "dbo.isjson" could not be found.
```
And if the function call is not schema-qualified, then it will use the system function and not the user-defined function. These changes also reflect this behavior. I can add tests to validate this functionality as well.

Reply:
Actually I was thinking without specifying the schema in function and relying on search path. for example
```
1> create database tdb
2> go
1> use tdb
2> go
Changed database context to 'tdb'.
1> CREATE FUNCTION isjson(@i text) returns int as begin return -99 end
2> go
1> create table #t1 (a int default isjson('1'), b int)
2> go
1> insert into #t1 (b) values (1)
2> go

(1 rows affected)
1> select * from #t1
2> go
a           b          
----------- -----------
        -99           1
```